### PR TITLE
Add Cross Cluster Parent Close Policy Types

### PR DESCRIPTION
**Pull request by demirkayaender**

### CODE REVIEW

File: thrift/shared.thrift [hunk 1]
```diff
@@ -1808,12 +1808,24 @@ struct CrossClusterRecordChildWorkflowExecutionCompleteRequestAttributes {
 struct CrossClusterRecordChildWorkflowExecutionCompleteResponseAttributes {
 }
 
+struct CrossClusterApplyParentClosePolicyRequestAttributes {
+  10: optional string targetDomainID
+  20: optional string targetWorkflowID
+  30: optional string targetRunID
+  40: optional i64 (js.type = "Long") initiatedEventID
```
Comment (yux0) [inline thrift/shared.thrift:1815]: What is this initiated event id used for?

**Comment by yux0** [comment]
LGTM. We have two approaches to handle parent close policy, workflow or direct call. I assume we just support the direct call approach. Is this correct?

File: thrift/shared.thrift [hunk 2]
```diff
@@ -1808,12 +1808,24 @@ struct CrossClusterRecordChildWorkflowExecutionCompleteRequestAttributes {
 struct CrossClusterRecordChildWorkflowExecutionCompleteResponseAttributes {
 }
 
+struct CrossClusterApplyParentClosePolicyRequestAttributes {
```
Comment (yycptt) [inline thrift/shared.thrift:1811]: looks like this attribute is the information for applying parent close policy for one workflow. Does it mean we will generate N x-cluster tasks for N childworkflows in the source cluster, even if their target cluster is the same. Or there will be only one x-cluster task, but when sending the request to the target cluster, we represent them using N attributes?

File: thrift/shared.thrift [hunk 3]
```diff
@@ -1808,12 +1808,27 @@ struct CrossClusterRecordChildWorkflowExecutionCompleteRequestAttributes {
 struct CrossClusterRecordChildWorkflowExecutionCompleteResponseAttributes {
 }
 
+struct AppyParentClosePolicyAttributes {
+  10: optional string domainID
+  20: optional string workflowID
+  30: optional string runID
```
[the remark this answers is not in the document]
Comment (demirkayaender) [reply thrift/shared.thrift:1814]: Yeah, target or child can make it easier to understand, let me change.

**Review by yycptt — APPROVED** [review]

**Pull request merged**